src/main/42-apigw.tf

```diff
@@ -0,0 +1,94 @@
+locals {
+  openapi_spec = var.openapi_spec_path
+}
+
+resource "aws_api_gateway_rest_api" "apigw" {
+  name        = "${var.app_name}-apigw-${var.env}"
+  description = "${var.app_name} API Gateway"
+  body        = file(local.openapi_spec)
+}
+
+resource "aws_api_gateway_deployment" "deployment" {
+  rest_api_id = aws_api_gateway_rest_api.apigw.id
+
+  triggers = {
+    redeployment = sha1(jsonencode(aws_api_gateway_rest_api.apigw.body))
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_api_gateway_stage" "stage" {
+  deployment_id        = aws_api_gateway_deployment.deployment.id
+  rest_api_id          = aws_api_gateway_rest_api.apigw.id
+  stage_name           = var.env
+  xray_tracing_enabled = true
+  variables = {
+    aws_lb_nlb_dns_name                 = trimspace(aws_lb.nlb.dns_name)
+    aws_api_gateway_vpc_link_backend_id = trimspace(aws_api_gateway_vpc_link.backend.id)
+  }
+}
+
+resource "aws_api_gateway_method_settings" "settings" {
+  rest_api_id = aws_api_gateway_rest_api.apigw.id
+  stage_name  = aws_api_gateway_stage.stage.stage_name
+  method_path = "*/*"
+
+  settings {
+    metrics_enabled    = true
+    logging_level      = "OFF"
+    data_trace_enabled = true
```

Comment:
Do you have any reference for this setting? I can't find it in the documentation, and I don't see this setting in the web console 🤔 It looks unrelated to the X-Ray setting. 
If you can't find any documentation on this, you can remove it from here 👍

Reply:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_method_settings#data_trace_enabled.

You're right it's not XRAY related. The XARY setting is already enabled in the `stage` definition. I'll remove it